### DISCUSSION TITLE
Work with Pandas DF

### DISCUSSION THREAD

**Discussion by Yanni8** [microsoft/autogen#1503] [Q&A]
Does anyone already have experience with working with Pandas Data frames?
I'm looking for something similar, like the [Lang Chain Pandas DF](https://python.langchain.com/docs/integrations/toolkits/pandas) Tool. 

Does AutoGen have a similar functionality (beside the fact that it can execute Python code)? 
One option would be to define a custom Skill (Tool). Has someone already tried this, and if yes, how did it go?

**Comment by ekzhu** [comment]
We will add an IPython code execution environment that is useful for data science. #1405

Meanwhile if you know exactly what you want to do with a dataframe, you can define a set of functions that operate on a dataframe and register them to the agent using `autogen.agentchat.register_function`.

**Comment by mraguth** [comment]
Able to setup 'pandasai' agent via writing skill (pandasai - chat function) in Autogen studio. Seems like working fine.

**Reply by sonichi** [reply]
Great. It'll be nice to hear your solution.
